test/notebook_tests/test_notebooks.py

```diff
@@ -52,15 +41,11 @@
 
 
 def _check_exclusive_device_availability(notebook_path, unavailable_devices):
-    if notebook_path.split("/")[-1] not in NOTEBOOKS_REQUIRING_DEVICE_AVAILABILITY:
-        return "None", True
-
     with open(notebook_path) as file:
-        for line in file:
-            for device in unavailable_devices:
-                if device.arn in line:
-                    return device.arn, False
-
+        content = file.read()
+        for device in unavailable_devices:
```

Comment:
I agree, buffer would be required in future if the notebooks is being executed and device becomes available next minute. But that would be a race condition kind of thing.

Reply:
The problem is that it would get a lot more complicated, because we just look at availability times, we don't actually forecast when it's available. 

It would be quite a change to try and forecast how long the notebook would take to run and figure out if the device will be available - and it wouldn't quite work if devices are taken offline for other reasons.